src/server/src/main/java/io/cassandrareaper/storage/CassandraStorage.java

```diff
@@ -390,18 +398,23 @@ public RepairRun addRepairRun(Builder repairRun, Collection<RepairSegment.Builde
               segment.getFailCount()));
       } else {
 
-        repairRunBatch.add(
-            insertRepairSegmentPrepStmt.bind(
-              segment.getRunId(),
-              segment.getId(),
-              segment.getRepairUnitId(),
-              segment.getStartToken(),
-              segment.getEndToken(),
-              segment.getState().ordinal(),
-              segment.getFailCount()));
+        try {
+          repairRunBatch.add(
+              insertRepairSegmentPrepStmt.bind(
+                  segment.getRunId(),
+                  segment.getId(),
+                  segment.getRepairUnitId(),
+                  segment.getStartToken(),
+                  segment.getEndToken(),
+                  segment.getState().ordinal(),
+                  segment.getFailCount(),
+                  objectMapper.writeValueAsString(segment.getTokenRange().getTokenRanges())));
+        } catch (JsonProcessingException e) {
+          throw new ReaperException(e);
+        }
       }
 
-      if (100 <= repairRunBatch.size()) {
+      if (100 <= repairRunBatch.size() || segment.getTokenRange().getTokenRanges().size() > 1) {
```

Comment:
why does a segment with more than one token range warrant a new batch?

Reply:
Batches were failing consistently on large clusters because of the token ranges set. I decided to process segments that have several token ranges as the only query in the batch to prevent such failures.

Reply:
addressed here: https://github.com/thelastpickle/cassandra-reaper/pull/373/files#diff-395e506278d41989792e2eb5585367c4R421